### DISCUSSION TITLE
테스트 커버리지가 높다고 무조건 좋은 건가요?

### DISCUSSION THREAD

**Discussion by le2sky** [2024-woowacourse-study/level-interview#45] [Test]
.

**Comment by J-I-H-O** [comment]
테스트 커버리지가 높다고 해서 무조건 좋은 것은 아닙니다.
먼저 테스트 커버리지란, 테스트를 통해 얼마나 많은 코드가 실행되었는지를 측정하는 지표입니다.
테스트 커버리지가 높다는 것은 많은 코드가 실행되었음을 의미하나, 많은 코드가 실행되었다고 해서 테스트가 완벽하다는 것을 의미하지는 않습니다.
또, 테스트 커버리지가 높다고 해서 테스트의 품질이 보장되는 것은 아니며, 테스트 커버리지를 높이기 위해 불필요한 테스트를 작성하거나 중복된 테스트를 실행하는 경우에는 오버헤드가 발생할 수 있습니다.

**Comment by 3Juhwan** [comment]
테스트 커버리지가 높다는 것은 코드의 일부가 테스트에 의해 실행되어 검증되었다는 것을 의미합니다. 높은 테스트 커버리지는 일반적으로 코드의 안정성과 신뢰성을 높일 수 있지만, 이것이 무조건적으로 항상 좋다고 말할 수는 없습니다.

테스트 커버리지가 높다고 해서 모든 버그를 찾아낼 수 있는 것은 아닙니다. 테스트 커버리지가 높더라도 테스트 케이스가 부족하거나 부적절하게 작성되었다면 여전히 중요한 버그가 발생할 수 있습니다. 또한, 모든 코드를 테스트하는 것이 현실적이지 않을 수도 있습니다. 특히 예외 상황이나 경계 조건을 모두 다루는 것은 어려울 수 있습니다.

**Comment by GIVEN53** [comment]
테스트가 작성되지 않은 라인, 분기를 확인하는 지표로 활용
발생할 수 있는 다양한 테스트 케이스를 작성하는 것이 중요
